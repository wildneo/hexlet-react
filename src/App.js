import React from 'react';
// import BtnGroup from './BtnGroup/BtnGroup';
// import Carousel from './Carousel/Carousel';
// import Collapse from './Collapse/Collapse';
import MyForm from './MyForm/MyForm';
import './App.css';
// import logo from './logo.svg';

function App() {
  // const images = ['/images/first.jpeg', '/images/second.jpeg', '/images/third.jpeg'];
  // const text = 'collapse me';

  return (
    <>
      <div className="container m-3">
        <MyForm />
      </div>
      {/* <div className="container m-3">
        <BtnGroup />
      </div>
      <div className="container m-3">
        <Carousel images={images} />
      </div>
      <div className="container m-3">
        <Collapse text={text} />
      </div> */}
    </>
  );
}

export default App;
