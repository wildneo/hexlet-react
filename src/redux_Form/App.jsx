import React from 'react';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleClick(event) {
    event.preventDefault();
    const { dispatch, reset } = this.props;
    dispatch(reset());
  }

  handleChange(event) {
    event.preventDefault();
    const { dispatch, change } = this.props;
    const { target: { value } } = event;
    dispatch(change(value));
  }

  render() {
    const { text } = this.props;

    return (
      <div>
        <form>
          <input onChange={this.handleChange} type="text" value={text} />
          <button onClick={this.handleClick} type="button">Reset</button>
        </form>
        {text && <div>{text}</div>}
      </div>
    );
  }
}

App.defaultProps = {
  text: '',
};
