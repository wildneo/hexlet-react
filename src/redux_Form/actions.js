export const change = (value) => ({
  type: 'TEXT_CHANGE',
  payload: {
    value,
  },
});

export const reset = () => ({
  type: 'TEXT_RESET',
});
