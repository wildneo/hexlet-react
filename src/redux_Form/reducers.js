import { combineReducers } from 'redux';

const text = (state = '', action) => {
  switch (action.type) {
    case 'TEXT_CHANGE':
      return action.payload.value;

    case 'TEXT_RESET': {
      return '';
    }

    default:
      return state;
  }
};

export default combineReducers({
  text,
});
