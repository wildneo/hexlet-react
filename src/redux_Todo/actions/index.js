export const updateNewTaskText = (text) => ({
  type: 'TEXT_UPDATE',
  payload: {
    text,
  },
});

// BEGIN (write your solution here)

// END
