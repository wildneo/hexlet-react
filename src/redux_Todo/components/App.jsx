import React from 'react';
import NewTaskForm from './NewTaskForm';
import Panel from './Panel';
import Tasks from './Tasks';

export default () => (
  <div className="col-5">
    <NewTaskForm />
    <Panel />
    <Tasks />
  </div>
);
