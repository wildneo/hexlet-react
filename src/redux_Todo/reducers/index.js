import { combineReducers } from 'redux';

const text = (state = '', action) => {
  switch (action.type) {
    case 'TEXT_UPDATE':
      return action.payload.text;

    case 'TASK_ADD':
      return '';

    default:
      return state;
  }
};

const tasks = (state = [], action) => {
  switch (action.type) {
    case 'TASK_ADD': {
      const { task } = action.payload;
      return [task, ...state];
    }

    case 'TASK_REMOVE': {
      const { id } = action.payload;
      return state.filter((task) => task.id !== id);
    }

    case 'TASK_CLEAN':
      return [];

    default:
      return state;
  }
};

export default combineReducers({
  text,
  tasks,
});
