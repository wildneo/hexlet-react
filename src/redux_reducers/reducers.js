import _ from 'lodash';
import { combineReducers } from 'redux';

const tasks = (state = {}, action) => {
  switch (action.type) {
    case 'TASK_ADD': {
      const { task } = action.payload;

      return { ...state, [task.id]: task };
    }

    case 'TASK_REMOVE': {
      const { id } = action.payload;

      return _.omit(state, id);
    }

    default:
      return state;
  }
};

const comments = (state = {}, action) => {
  switch (action.type) {
    case 'TASK_REMOVE': {
      const { id } = action.payload;

      return _.omitBy(state, (c) => c.taskId === id);
    }

    case 'TASK_COMMENT_ADD': {
      const { comment } = action.payload;

      return { ...state, [comment.id]: comment };
    }

    case 'TASK_COMMENT_REMOVE': {
      const { id } = action.payload;

      return _.omit(state, id);
    }

    default:
      return state;
  }
};

export default combineReducers({ tasks, comments });
