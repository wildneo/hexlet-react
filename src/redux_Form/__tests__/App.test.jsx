import React from 'react';
import { createStore } from 'redux';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import reducers from '../reducers';
import App from '../App';
import * as actionCreators from '../actions';

Enzyme.configure({ adapter: new Adapter() });

test('Store', () => {
  const store = createStore(reducers);

  // eslint-disable-next-line react/jsx-props-no-spreading
  const vdom = <App dispatch={store.dispatch} {...actionCreators} />;
  const wrapper = mount(vdom);
  store.subscribe(() => {
    const { text } = store.getState();
    wrapper.setProps({ text });
  });

  expect(wrapper.render()).toMatchSnapshot();

  const newTextInput = wrapper.find('input[type="text"]');
  const newTextReset = wrapper.find('button');

  newTextInput.simulate('change', { target: { value: 'na-na' } });
  wrapper.update();
  expect(wrapper.render()).toMatchSnapshot();

  newTextReset.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  newTextInput.simulate('change', { target: { value: 'another task' } });
  expect(wrapper.render()).toMatchSnapshot();

  newTextReset.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
