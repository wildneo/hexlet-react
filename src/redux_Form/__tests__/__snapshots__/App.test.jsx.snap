// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store 1`] = `
<div>
  <form>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Store 2`] = `
<div>
  <form>
    <input
      type="text"
      value="na-na"
    />
    <button
      type="button"
    >
      Reset
    </button>
  </form>
  <div>
    na-na
  </div>
</div>
`;

exports[`Store 3`] = `
<div>
  <form>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;

exports[`Store 4`] = `
<div>
  <form>
    <input
      type="text"
      value="another task"
    />
    <button
      type="button"
    >
      Reset
    </button>
  </form>
  <div>
    another task
  </div>
</div>
`;

exports[`Store 5`] = `
<div>
  <form>
    <input
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Reset
    </button>
  </form>
</div>
`;
